crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -757,10 +757,18 @@ impl<F: Send + Clone + Sync> GetTracker<F, PaymentData<F>, api::PaymentsRequest>
         });
 
         let order_tax_amount = pmt_order_tax_amount.or_else(|| {
-            payment_intent
-                .tax_details
-                .clone()
-                .and_then(|tax| tax.default.map(|a| a.order_tax_amount))
+            payment_intent.tax_details.as_ref().and_then(|tax_details| {
+                tax_details
+                    .payment_method_type
+                    .as_ref()
+                    .map(|payment_method_tax| payment_method_tax.order_tax_amount)
```

Comment:
@swangi-kumari we should validate here that the payment method that is passed in the request matches with the payment method that for which the tax was calculated right?

Reply:
This validation was already done in `pmt_order_tax_amount`. Why are fetching `order_tax_amount` again?

Reply:
This validation for payment_method_type is already done in `pmt_order_tax_amount`.  We don't need these changes.